### PR TITLE
[Android] Implement swiping effects.

### LINKED ISSUE #264: [Android] Swipe left/right animations
- [x] Swipe to complete doesn't make the bar green
- [x] Swiping items left/right cuts doesn't complete the animation, .e.g Swiping right to mark as complete. When you lift your finger, the items just snaps to its final state. Instead it should complete the animation out of the screen before marking it as complete

### PR DESCRIPTION
**Pull request by dalinaum**
* Swipe to complete make the bar green
* Swiping items left / right invoke their animations.

Fix #264, please review this @realm/java

### CODE REVIEW

File: RealmTasks Android/app/src/main/java/io/realm/realmtasks/list/ItemViewHolder.java [hunk 1]
```diff
@@ -191,12 +199,16 @@ public void setIconBarAlpha(float alpha) {
         iconBar.setAlpha(alpha);
     }
 
-    public void preCompleted() {
+    public void changeBackgroundColorIfNeeded() {
+        if (backgroundColorVersion == completedVersion) {
+            return;
+        }
         if (getCompleted()) {
```
Comment (nhachicha) [inline RealmTasks Android/app/src/main/java/io/realm/realmtasks/list/ItemViewHolder.java]: you can access the boolean directly for speed like you did in https://github.com/realm/RealmTasks/blob/6e4f8ac6d73e49712855769bcd69ec37bce5a6e7/RealmTasks%20Android/app/src/main/java/io/realm/realmtasks/list/ItemViewHolder.java#L77

**Comment by dalinaum** [comment]
Updated. @nhachicha @realm/java

**Comment by dalinaum** [comment]
Updated the PR as reviews.

**Comment by dalinaum** [comment]
I'll close this, please review #321

**Pull request closed without merging**